pydantic/_internal/_std_types_schema.py

```diff
@@ -307,6 +307,7 @@ class SequenceValidator:
     min_length: int | None = None
     max_length: int | None = None
     strict: bool | None = None
+    fail_fast: bool | None = None
```

Comment:
Same question - is `fail_fast` applicable to all sequence types? It probably should be for this case, but the case about where we reuse `SEQUENCE_CONSTRAINTS` is where more of my concern lies.

Reply:
Also, let's add some tests for `my_field: Sequence = Field(..., fail_fast=True)`, etc.

Reply:
Then see what happens with a `deque` or something like that

Reply:
> Same question - is fail_fast applicable to all sequence types? 

You are right, it's not applicable to all sequence types. But I don't know how to correctly pass fail_fast to schema. Before calling `apply_known_metadata` it' always calls `sequence_like_prepare_pydantic_annotations` that fails because `fail_fast` is not in a list of `SEQUENCE_CONSTRAINTS`.

Maybe you have PR example how to correctly implement staff like this?

Reply:
> Also, let's add some tests for my_field: Sequence = Field(..., fail_fast=True), etc.

It won't work for now. I guess first we need to implement a way how from python say rust to stop validation. 

More context here - https://github.com/pydantic/pydantic-core/pull/1322#issuecomment-2166798608